GitExtUtils/DebugHelpers.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System.Diagnostics;
+
+namespace GitExtUtils;
+
+/// <summary>
+///  Set of DEBUG-only helpers.
+/// </summary>
+public static class DebugHelpers
+{
+    [Conditional("DEBUG")]
+    public static void Assert(bool condition, string message)
+    {
+        if (Debugger.IsAttached)
+        {
+            Debug.Assert(condition, message);
+        }
+        else if (!condition)
+        {
+            Debugger.Launch();
+
+            if (!Debugger.IsAttached)
+            {
+                throw new InvalidOperationException(message);
+            }
+        }
```

Comment:
```suggestion
        if (!condition)
        {
            Fail(message);
        }
```

Reply:
Actually, thinking about this, I don't think that's the expected or desired behaviour. `Debug.Assert` allows skipping over the check, whereas `Debug.Fail` results in the only choice - the end.

Reply:
> Actually, thinking about this, I don't think that's the expected or desired behaviour. `Debug.Assert` allows skipping over the check, whereas `Debug.Fail` results in the only choice - the end.

I doubt there is a different behavior. According to SourceLink, it is implemented as I expect which is "the same way":

```
        [Conditional("DEBUG")]
        public static void Assert([DoesNotReturnIf(false)] bool condition, string? message, string? detailMessage)
        {
            if (!condition)
            {
                Fail(message, detailMessage);
            }
        }
```

"skipping over the check" is achieved by `[Conditional("DEBUG")]` (if you meant that).

Reply:
Thank you for linking me to the source.